map/framework.cpp

```diff
@@ -2585,11 +2585,7 @@ bool Framework::ParseDrapeDebugCommand(string const & query)
 
   if (desiredStyle != MapStyleCount)
   {
-#if defined(OMIM_OS_ANDROID)
-    MarkMapStyle(desiredStyle);
```

Comment:
Why it was done only for Android before? What is the impact of this change on iOS?

Reply:
I suspect that it was made for Android with OGL. Where context and surface are re-created every time with Activity.
And it should stay for OGL.

Reply:
Yeap, it looks like for OGL everything is reset and updated after closing the search screen, but not for Vulkan.

```
void Framework::MarkMapStyle(MapStyle mapStyle)
{
  // In case of Vulkan rendering we don't recreate geometry and textures data, so
  // we need use SetMapStyle instead of MarkMapStyle in all cases.
  if (m_vulkanContextFactory)
    m_work.SetMapStyle(mapStyle);
  else
    m_work.MarkMapStyle(mapStyle);
}
```

@vng what's wrong if SetMapStyle() is used for OGL too?

Reply:
I'm not an expert here. The primary and key difference between Vulkan vs OGL (and the reason why we have 2 renderers) is context/surface/back buffer lifetime. Vulkan keeps them alive, while OGL recreates on every switch between Activity/Fragment. So the answer is somewhere here.

You can set OGL renderer and try SetMapStyle. But even if it works on your device, it may fail on some other one. GPUs and drivers routine in Android world is very complicated.

Reply:
@pastk can you minimize your changes by scoping them for android-only, vulkan-only?